ComponentKit/Utilities/CKComponentAction.mm

```diff
@@ -254,23 +247,32 @@ CKComponentViewAttributeValue CKComponentActionAttribute(const CKTypedComponentA
 
 @implementation CKComponentActionControlForwarder
 {
-  CKTypedComponentAction<UIEvent *> _action;
+  UIControlEvents _controlEvents;
 }
 
-- (instancetype)initWithAction:(const CKTypedComponentAction<UIEvent *> &)action
+- (instancetype)initWithControlEvents:(UIControlEvents)controlEvents
 {
   if (self = [super init]) {
-    _action = action;
+    _controlEvents = controlEvents;
   }
   return self;
 }
 
 - (void)handleControlEventFromSender:(UIControl *)sender withEvent:(UIEvent *)event
 {
-  // If the action can be handled by the sender itself, send it there instead of looking up the chain.
-  _action.send(sender.ck_component, CKComponentActionSendBehaviorStartAtSender, event);
+  CKComponentActionList *const list = objc_getAssociatedObject(sender, ck_actionListKey);
+  CKCAssertNotNil(list, @"Forwarder should always find an action list installed by applicator");
+  // Protect against mutation-during-enumeration by copying the list of actions to send:
```

Comment:
Yeah I think it's fine, I was just trying to imagine a case where this could happen. The only thing I could think of is someone calling cellFor... on the collection view, which I think could theoretically happen.